plugins/notifications-backend/src/service/router.ts

```diff
@@ -1,61 +1,160 @@
 import { Logger } from 'winston';
+import { Config } from '@backstage/config';
 
 export interface RouterOptions {
   logger: Logger;
-  dbClient?: Knex<any, any[]>;
+  dbConfig: Config;
+  catalogClient: CatalogClient;
 }
 
+export type CreateRequest = {
+  subject: string;
+  body?: string;
+  targetUsers?: string[];
+  targetGroups?: string[];
+}
+
+
+
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, dbClient } = options;
+  const { logger, dbConfig, catalogClient } = options;
 
   const router = Router();
   router.use(express.json());
 
-  if (!dbClient) {
-    logger.error('Missing dbClient');
+  if (!dbConfig) {
+    logger.error('Missing dbConfig');
     return router;
   }
 
-  router.get('/health', (_, response) => {
-    logger.info('PONG!');
-    response.json({ status: 'ok' });
+  const dbClient = knex({
+    client: dbConfig.getString('client'),
+    connection: {
+      host: dbConfig.getString('connection.host'),
+      port: dbConfig.getNumber('connection.port'),
+      user: dbConfig.getString('connection.user'),
+      password: dbConfig.getString('connection.password'),
+      database: 'backstage_plugin_notifications'
+    }
+  });
+
+  if (! await dbClient.schema.hasTable('messages')) {
+    await dbClient.schema.createTable('messages', function (table) {
+      table.uuid("id", {primaryKey: true}).defaultTo(dbClient.fn.uuid());
+      table.string('subject');
+      table.string('body');
+    })
+
+  }
+
+  router.get('/health', async (_, response) => {
+    var myRes = 'ok';
+
+    const userRef = await catalogClient.getEntityByRef("user:jdoe");
+
+    if (!userRef) {
+      myRes = 'fail'
+    } else {
+      if (userRef.relations && userRef.relations.length > 0) {
+        userRef.relations.forEach((relation)=>{
+          const parts = relation.targetRef.split(':');
+          if (parts.length == 0 || parts[0] != 'group') {
+            return;
+          }
+          myRes = `${myRes}:${parts[1]}`;
+        });
+      }
+    }
+    
+    response.json({ status: myRes });
   });
 
-  router.get('/notifications/count', async (_, response) => {
+  router.get('/notifications/count/:text?', async ({params: {text}}, response) => {
     let msgcount = -1;
-    await dbClient('messages')
-      .count('* as CNT')
-      .then(count => {
+    
+    var query = dbClient('messages').count('* as CNT');
+    
+    if ( text && text.length > 0 ) {
+      query.whereILike('subject', `%${text}%`).orWhereILike('body', `%${text}%`)
+    }
+    
+    await query.then(count => {
         msgcount = (count[0]?.CNT as unknown as number) || -1;
       });
+
     response.json({ status: 'ok', count: msgcount });
   });
 
   router.post('/notifications', async (request, response) => {
     let msgid = -1;
+
+    var createRequest = request.body as CreateRequest;
+    
+    if (createRequest.targetGroups) {
+      createRequest.targetGroups.forEach(async group => {
```

Comment:
Can be `string.forEach` ==> error

Reply:
Oh, sorry. I meant that the `forEach` can be called on something else than an array as expected. So it would be an error. But this line is outdated already.